src/Scripts/Ecs/Components/AttackComponent.gd

```diff
@@ -0,0 +1,6 @@
+extends "res://Scripts/Ecs/Core/Component.gd"
+
```

Comment:
I would rename this to something like `CombatComponent`. One way around the event bus is to use an `AttackIntent` component and have a system process just those, in order (input-handler system / AI system makes them).